src/notary/mod.rs

```diff
@@ -1,24 +1,118 @@
-use elliptic_curve::pkcs8::DecodePrivateKey;
-use futures::{AsyncRead, AsyncWrite};
-use tlsn_verifier::tls::{Verifier, VerifierConfig};
-
-/// Default for the maximum number of bytes that can be sent (8Kb).
-pub const DEFAULT_MAX_SENT_LIMIT: usize = 1 << 12;
-/// Default for the maximum number of bytes that can be received (8Kb).
-pub const DEFAULT_MAX_RECV_LIMIT: usize = 1 << 14;
-
-/// Runs a simple Notary with the provided connection to the Prover.
-pub async fn run_notary<T: AsyncWrite + AsyncRead + Send + Unpin + 'static>(conn: T) {
-    // Load the notary signing key
-    let signing_key_str = std::str::from_utf8(include_bytes!("../fixture/notary.key")).unwrap();
-    let signing_key = p256::ecdsa::SigningKey::from_pkcs8_pem(signing_key_str).unwrap();
-
-    // Setup default config. Normally a different ID would be generated
-    // for each notarization.
-    let config = VerifierConfig::builder().id("example").build().unwrap();
-
-    Verifier::new(config)
-        .notarize::<_, p256::ecdsa::Signature>(conn, &signing_key)
+use std::sync::Arc;
+
+use http_body_util::{BodyExt as _, Either, Empty, Full};
+use hyper::{client::conn::http1::Parts, Request, StatusCode};
+use hyper_util::rt::TokioIo;
+use notary_server::{ClientType, NotarizationSessionRequest, NotarizationSessionResponse};
+use rustls::{Certificate, ClientConfig, RootCertStore};
+use tokio::net::TcpStream;
+use tokio_rustls::TlsConnector;
+use tokio_util::bytes::Bytes;
+
+/// Requests notarization from the Notary server.
+pub async fn request_notarization(
+    host: &str,
+    port: u16,
+    max_sent_data: Option<usize>,
+    max_recv_data: Option<usize>,
```

Comment:
Remove these params and move predefined const from `notarize.rs` here

Reply:
Changed in latest commit